app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    result_args_dict = {}
+
+    def inner(*args) -> Callable:
+        if args in result_args_dict:
+            print("Getting from cache")
+        else:
+            result_args_dict[args] = func(*args)
+            print("Calculating new result")
+            return result_args_dict[args]
+        return result_args_dict[args]
```

Comment:
Add a code suggestion made by @pavlejviki, and remove line 13 from your code. There is no need for code duplication for these two return statements.